src/python/doc/persistent_cohomology_sum.inc

```diff
@@ -21,4 +21,5 @@
    |                                                                 | * :doc:`simplex_tree_ref`                                                                                             |
    |                                                                 | * :doc:`cubical_complex_ref`                                                                                          |
    |                                                                 | * :doc:`periodic_cubical_complex_ref`                                                                                 |
+   |                                                                 | * :doc:`edge_collapse`                                                                                                |
```

Comment:
I am not sure of where we should mention this new module. In C++, we have:

![image](https://github.com/GUDHI/gudhi-devel/assets/10407034/8391acda-2e9d-4a93-9aea-cbdbb4fef6d3)

To do something equivalent for the python version, you can add in `src/python/doc/index.rst`, between Rips and Witness complex:
```rst

Edge collapse
=============

.. table::
   :widths: 30 40 30

   +-----------------------------------------------------------------+---------------------------------------------------------------------+-----------------------------------------------+
   | .. figure::                                                     | Edge collapse is able to reduce any flag filtration to a smaller    | :Author: Siddharth Pritam, Marc Glisse        |
   |      ../../doc/Collapse/dominated_edge.png                      | flag filtration with the same persistence, using only the           |                                               |
   |      :figclass: align-center                                    | 1-skeletons of a simplicial complex.                                | :Since: GUDHI 3.9.0                           |                                         
   |                                                                 | The reduction is exact and the persistence homology of the reduced  |                                               |
   |                                                                 | sequence is identical to the persistence homology of the input      |                                               |
   |                                                                 | sequence. The resulting method is simple and extremely efficient.   |                                               |
   |                                                                 |                                                                     |                                               |
   |                                                                 | Computation of edge collapse and persistent homology of a filtered  |                                               |
   |                                                                 | flag complex via edge collapse as described in                      |                                               |
   |                                                                 | :cite:`edgecollapsearxiv`.                                          |                                               |
   |                                                                 |                                                                     |                                               |
   +-----------------------------------------------------------------+---------------------------------------------------------------------+-----------------------------------------------+
   |                                                                 |  * :doc:`edge_collapse`                                                                                             |
   +-----------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------+
```

Reply:
I moved it where you said. But (also in C++) it is a strange place. ("Basic operation: contraction" is also in a strange place)
Bah, we can always move it later...